### PR TITLE
TER-99 external buckets for TD

### PR DESCRIPTION
**Pull request by jirkasemmler**
Jira: TER-99

Common: https://github.com/keboola/php-storage-driver-common/pull/32
TD driver: https://github.com/keboola/php-storage-driver-teradata/pull/55
Connection: https://github.com/keboola/connection/pull/4160
SAPI https://github.com/keboola/storage-api-php-client/pull/981

Before asking for review make sure that:

## Checklist

- [x] New client method(s) has tests
- [x] Apiary file is updated

## Release

  - [x] I gave the PR a proper label:
    * major (BC break)
    * minor (new feature)
    * patch (backwards compatible fix)
    * no release (just test changes)

### CODE REVIEW

**Review by romanbracinik — COMMENTED** [review]
Otestestoval som to, rucne registrovanie bucketu a refresh funguje  dva testiky mi ale lokalne spadnu.
Screenshot 2023-01-24 at 10 21 57
Ked to pustim prvi krat tak nesedi assert, ked to spustim potom znova tak to padne na tom ze bucket uz existuje
Screenshot 2023-01-24 at 10 40 58

A potom mi pada este jeden test priamo na TD to uz asi moze byt neco blbe ako som si registroval novy TD backend alebo neviem 
Screenshot 2023-01-24 at 10 23 02

File: tests/Backend/ExternalBuckets/TeradataRegisterBucketTest.php [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+        $runId = $this->setRunId();
+        $idOfBucket = $this->_client->registerBucket(
+            'test-bucket-registration',
+            [$workspace['connection']['schema']],
+            'in',
+            'Iam in workspace',
+            $this->thisBackend,
+            'Iam-your-workspace'
+        );
+        $this->assertEvents($runId, ['storage.bucketCreated']);
```
Comment (romanbracinik) [inline tests/Backend/ExternalBuckets/TeradataRegisterBucketTest.php:65]: uz mozes pouzivat nove assertovanie event 
```
$this->initEvents($this->_client);
 $assertCallback = function ($events) {
            $this->assertCount(1, $events);
        };
        $query = new EventsQueryBuilder();
        $query->setEvent('storage.bucketCreated')
            ->setRunId($runId);
        $this->assertEventWithRetries($this->_client, $assertCallback, $query);
```
https://github.com/keboola/storage-api-php-client/pull/979/files

Comment (jirkasemmler) [reply tests/Backend/ExternalBuckets/TeradataRegisterBucketTest.php:69]: je to tam :)

Comment (romanbracinik) [reply tests/Backend/ExternalBuckets/TeradataRegisterBucketTest.php:69]: A tu na riadku 69 to nejde pouzit ?

**Comment by jirkasemmler** [comment]
@romanbracinik uz tam jsou ty nove eventy, naslo mi to bug v connection, kterej jsem fixnul tady https://github.com/keboola/connection/pull/4160/commits/47d13f90905586c34cd65f2cae6e70fb9b79848d (ono to fungovalo, ale nestrilelo eventy) . prosim o posledni kolecko at to muze ven

File: tests/Backend/ExternalBuckets/BaseExternalBuckets.php [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    {
+        $runId = $this->_client->generateRunId();
+        $this->_client->setRunId($runId);
+        return $runId;
+    }
+
+    /**
+     * @param string[] $expectedEventsNames
+     */
+    protected function assertEvents(string $runId, array $expectedEventsNames): void
```
Comment (romanbracinik) [inline tests/Backend/ExternalBuckets/BaseExternalBuckets.php:59]: toto bude teraz nepouzivana metoda

File: tests/Backend/ExternalBuckets/TeradataRegisterBucketTest.php [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+use Keboola\Test\Backend\Workspaces\Backend\TeradataWorkspaceBackend;
+use Keboola\Test\Backend\Workspaces\Backend\WorkspaceBackendFactory;
+use Keboola\Test\Utils\EventsQueryBuilder;
+
+class TeradataRegisterBucketTest extends BaseExternalBuckets
+{
+    public function setUp(): void
+    {
+        parent::setUp();
+        $this->initEvents($this->_client);
```
Comment (romanbracinik) [inline tests/Backend/ExternalBuckets/TeradataRegisterBucketTest.php:18]: toto volas v kazdom teste na pozadovanom mieste tu to imho teda nemusi byt, zbytocne sa 2x zaklada dummy eventa

**Review by romanbracinik — CHANGES_REQUESTED** [review]
dve drobnosti

**Review by romanbracinik — APPROVED** [review]
👍

**Pull request merged**